scripts/proteinInteractionMINT/parse_mint.py

```diff
@@ -0,0 +1,418 @@
+        key = ali_split[1][:-1]
+        value = ali_split[0].split(':')[1]
+        alias_map[key] = value
+    if 'display_long' in alias_map:
+        dcid = alias_map['display_long']
+        if re.search('[\W]+', dcid) or len(dcid.split('_')) != 2:
+            return False
+    return True
+
+def get_property_content(content, prefix):
+    """Add the prefix to each object in the content and return the
+    concatenated string with "," as the separator.
+    Args:
+        content: the list containing property objects
+        prefix: the prefix before dcid, such as "dcs:"
+    Returns:
+        objects separated by comma. For example:
+        "dcs:bio/CWH41_YEAST,dcs:bio/RPN3_YEAST"
+    """
+    if not content:
+        return None
+    item_list = []
+    for obj in content:
+        item_list.append(prefix + obj)
+    return ','.join(item_list)
+
+def get_cur_line(key_name, value_list, prefix):
+    """Return the line of property schema from objects, property name and prefix
+    Args:
+        key_name: property name
+        value_list: object list
+        prefix: prefix for the objects
+    Return:
+        The property string line in a schema. For example:
+        "interactingProtein: dcs:bio/CWH41_YEAST,dcs:bio/RPN3_YEAST"
+    """
+    property_content = get_property_content(value_list, prefix)
+    if not property_content:
+        return None
+    cur_line = key_name + ': ' + property_content
+    return cur_line
+
+def get_schema_from_text(term, new_source_map, psimi_to_dcid):
+    """
+    Args:
+        term: a list with each item containing the information
+        new_source_map: a map contaning new source information. For example:
+        {"refereces":{},"identifier":{},"confidence":{"newConfidence":"AA10010"}}
+
+    Returns:
+        a string, which is a data schema
+        new_source_map: and a map with new reference,identifier and confidence sources. For example:
+
+        ['''Node: dcid:bio/MT1A_HUMAN_P53_HUMAN
+        typeOf: ProteinProteinInteraction
+        name: "MT1A_HUMAN_P53_HUMAN"
+        interactingProtein: dcs:bio/UniProt_MT1A_HUMAN,dcs:bio/UniProt_P53_HUMAN
+        interactionDetectionMethod: dcs:AntiBaitCoimmunoprecipitation
+        interactionType: dcs:PhysicalAssociation
+        interactionSource: dcs:Mint
+        intActID: "EBI-8045171"
+        mintID: "MINT-1781444"
+        imexID: "IM-11231-3"
+        confidenceScore: [0.54 dcs:IntactMiScore]
+        pubMedID: "16442532"
+        imexID: "IM-11231"
+        mintID: "MINT-5218281"''',{"references":{},"identifier":{},
+        "confidence":{"newConfidence":"AA10010"}}]
+    """
+    term_map = collections.defaultdict(list)
+    protein = get_protein_dcid(term[4])
+    if protein:
+        term_map['interactingProtein'].append(protein)
+    protein = get_protein_dcid(term[5])
+    if protein:
+        term_map['interactingProtein'].append(protein)
+    detection_method = psimi_to_dcid[term[6].split(':"')[1].split('(')[0][:-1]]
```

Comment:
given that this repeats 3 times, it could use a helper function, so its more understandable what happens from the name.

Reply:
Added the helper function.